File: apps/api/src/schemas/follow.ts
export const followingSchema = {
  $id: "/schemas/follow",
  type: "object",
  required: ["id", "status", "createdAt", "user", "followsBack"],
  properties: {
    id: { type: "number" },
    status: {
      type: "string",
      enum: ["pending", "following", "none"],
    },
    createdAt: { type: "string" },
    user: { $ref: "/schemas/user" },
    followsBack: {
      type: "string",
      enum: ["pending", "following", "none"],
    },
  },
};

File: src/peated/models/follow.py
from sqlalchemy import BigInteger, Column, DateTime, Enum, ForeignKeyConstraint, Index, PrimaryKeyConstraint, text

from peated.db.base import Base


class Follow(Base):
    __tablename__ = "follow"
    __table_args__ = (
        ForeignKeyConstraint(["from_user_id"], ["user.id"], name="follow_from_user_id_user_id_fk"),
        ForeignKeyConstraint(["to_user_id"], ["user.id"], name="follow_to_user_id_user_id_fk"),
        PrimaryKeyConstraint("id", name="follow_pkey"),
        Index("follow_unq", "from_user_id", "to_user_id", unique=True),
    )

    from_user_id = Column(BigInteger, nullable=False)
    to_user_id = Column(BigInteger, nullable=False)
    status = Column(
        Enum("none", "pending", "following", name="follow_status"),
        nullable=False,
        server_default=text("'pending'::follow_status"),
    )
    created_at = Column(DateTime, nullable=False, server_default=text("now()"))
    id = Column(BigInteger)
